Maple2.Server.World/Containers/PartyManager.cs

```diff
@@ -0,0 +1,216 @@
+            var request = new PartyRequest {
+                PartyId = Party.Id,
+                ReceiverIds = { player.CharacterId },
+                Invite = new PartyRequest.Types.Invite {
+                    SenderId = requestor.CharacterId,
+                    SenderName = requestor.Name,
+                },
+            };
+
+            PartyResponse response = client.Party(request);
+            return (PartyError) response.Error;
+        } catch (RpcException) {
+            return PartyError.s_party_err_not_found;
+        }
+    }
+
+    public PartyError Join(PlayerInfo info) {
+        if (Party.Members.Count >= Party.Capacity) {
+            return PartyError.s_party_err_full_limit_player;
+        }
+        if (Party.Members.ContainsKey(info.CharacterId)) {
+            return PartyError.s_party_err_alreadyInvite;
+        }
+
+        PartyMember member = new PartyMember {
+            PartyId = Party.Id,
+            Info = info.Clone(),
+            JoinTime = DateTimeOffset.UtcNow.ToUnixTimeSeconds(),
+            LoginTime = info.UpdateTime,
+        };
+
+        Broadcast(new PartyRequest {
+            AddMember = new PartyRequest.Types.AddMember {
+                CharacterId = member.CharacterId,
+                JoinTime = member.JoinTime,
+                LoginTime = member.LoginTime,
+            },
+        });
+        Party.Members.TryAdd(info.CharacterId, member);
+        return PartyError.none;
+    }
+
+    public PartyError Kick(long requestorId, long characterId) {
+        if (!Party.Members.TryGetValue(requestorId, out PartyMember? _)) {
+            return PartyError.s_party_err_not_found;
+        }
+        if (requestorId != Party.LeaderCharacterId) {
+            return PartyError.s_party_err_not_chief;
+        }
+        if (characterId == Party.LeaderCharacterId) {
+            return PartyError.none;
+        }
+        if (!Party.Members.TryGetValue(characterId, out PartyMember? member)) {
+            return PartyError.s_party_err_not_exist;
+        }
+        if (Party.Members.Count <= 2) {
+            Broadcast(new PartyRequest {
+                Disband = new PartyRequest.Types.Disband {
+                    CharacterId = characterId,
+                },
+            });
+            Dispose();
+            return PartyError.none;
+        }
+
+        Broadcast(new PartyRequest {
+            RemoveMember = new PartyRequest.Types.RemoveMember {
+                CharacterId = member.CharacterId,
+                IsKicked = true,
+            },
+        });
+        Party.Members.TryRemove(member.CharacterId, out _);
+        return PartyError.none;
+    }
+
+    public PartyError Leave(long characterId) {
+        if (!Party.Members.TryGetValue(characterId, out PartyMember? member)) {
+            return PartyError.s_party_err_not_found;
+        }
+
+        if (Party.Members.Count <= 2) {
+            Broadcast(new PartyRequest {
```

Comment:
might make sense to just move this `Broadcast(...)` into `Dispose()`. You're currently missing this Broadcast when the party is disbanded naturally.